tests/unit/tuner/test_onnx.py

```diff
@@ -0,0 +1,288 @@
+            torch.nn.Conv2d(2, 3, 2, stride=1),
+            torch.nn.ReLU(),
+        )
+
+    return _get_model
+
+
+@pytest.fixture
+def get_pytorch_lstm_model():
+    def _get_model(input_dim):
+        return torch.nn.Sequential(torch.nn.LSTM(32, 2))
+
+    return _get_model
+
+
+@pytest.fixture
+def get_keras_linear_model():
+    def _get_model(input_dim):
+        return keras.Sequential(
+            [
+                keras.layers.Flatten(input_shape=(input_dim,)),
+                keras.layers.Dense(64, activation='relu'),
+                keras.layers.Dense(32),
+            ]
+        )
+
+    return _get_model
+
+
+@pytest.fixture
+def get_keras_conv_model():
+    def _get_model(input_shape):
+        return keras.Sequential(
+            [
+                keras.layers.Input(input_shape),
+                keras.layers.Conv2D(input_shape[-1], 3, strides=2, activation='relu'),
+            ]
+        )
+
+    return _get_model
+
+
+@pytest.fixture
+def get_keras_lstm_model():
+    def _get_model(input_shape):
+        return keras.Sequential(
+            [
+                keras.layers.Input(input_shape),
+                keras.layers.LSTM(4),
+            ]
+        )
+
+    return _get_model
+
+
+@pytest.fixture
+def get_paddle_linear_model():
+    def _get_model(input_dim):
+        return paddle.nn.Sequential(
+            paddle.nn.Flatten(),
+            paddle.nn.Linear(in_features=input_dim, out_features=64),
+            paddle.nn.ReLU(),
+            paddle.nn.Linear(in_features=64, out_features=32),
+        )
+
+    return _get_model
+
+
+@pytest.fixture
+def get_paddle_conv_model():
+    def _get_model(input_shape):
+        return paddle.nn.Sequential(paddle.nn.Conv2D(3, 3, 2, stride=1))
+
+    return _get_model
+
+
+@pytest.fixture
+def get_paddle_lstm_model():
+    def _get_model():
+        return paddle.nn.Sequential(paddle.nn.LSTM(32, 2))
+
+    return _get_model
+
+
+def test_pytorch_to_onnx(get_pytorch_linear_model):
+    INPUT_SHAPE = [32]
+    temp_onnx_file = str(Path(tempfile.tempdir) / "finetuned.onnx")
+    model = get_pytorch_linear_model(32)
+    # convert to ONNX
+    to_onnx(model, temp_onnx_file, input_shape=INPUT_SHAPE)
+    validate_onnx_export(model, temp_onnx_file, INPUT_SHAPE)
+
+
+def test_pytorch_conv_to_onnx(get_pytorch_conv_model):
+    INPUT_SHAPE = [2, 32, 32]
+    temp_onnx_file = str(Path(tempfile.tempdir) / "finetuned.onnx")
+    model = get_pytorch_conv_model(INPUT_SHAPE)
+    to_onnx(model, temp_onnx_file, input_shape=INPUT_SHAPE)
+    # convert to ONNX
+    shape = [16] + INPUT_SHAPE
+    x = np.random.rand(*shape).astype(np.float32)
+    session = onnxruntime.InferenceSession(temp_onnx_file)
+
+    model.eval()
+    y_original = model(torch.Tensor(x)).detach().numpy()
+    y_exported = session.run(None, {session.get_inputs()[0].name: x})[0]
+
+    np.testing.assert_allclose(y_original, y_exported, rtol=1e-03, atol=1e-05)
+
+
+def test_pytorch_lstm_to_onnx(get_pytorch_lstm_model):
```

Comment:
@gmastrapas, contrary to what we spoke about, concerning putting the onnx evaluation into the onnx.py file, I put it into the test file. 
Even though it would be nice to have such functionality, as a user, the different types of models e.g. LSTM have outputs different from tensors. In case of lstm, it outputs the hidden state and the memory vector. 
We would need to handle all output types for all possible types of models. 
IMHO, that would be a little over-engineered and in the end it is only a few lines of code to evaluate the export, as a user.

Reply:
I think we can stick to embedding models, i.e. assume that all models return a batch of vectors `(batch_size, dim)`

Reply:
Solved in offline conversation. 
The evaluation is move back as George suggested, since we agreed it would be save to assume, that the embed model creates only one tensor as embedding for new.